examples/nlp/token_classification/punctuate_capitalize_infer.py

```diff
@@ -0,0 +1,739 @@
+            computation.
+    Returns:
+        A pytorch DataLoader.
+    """
+    if max_seq_length is None:
+        max_seq_length = model._cfg.dataset.max_seq_length
+    if step is None:
+        step = model._cfg.dataset.step
+    if margin is None:
+        margin = model._cfg.dataset.margin
+
+    dataset = BertPunctuationCapitalizationInferLongDataset(
+        tokenizer=model.tokenizer, queries=queries, max_seq_length=max_seq_length, step=step, margin=margin
+    )
+    return torch.utils.data.DataLoader(
+        dataset=dataset,
+        collate_fn=dataset.collate_fn,
+        batch_size=batch_size,
+        shuffle=False,
+        num_workers=model._cfg.dataset.num_workers,
+        pin_memory=model._cfg.dataset.pin_memory,
+        drop_last=False,
+    )
+
+
+def _move_acc_probs_to_token_preds(
+    pred: List[int], acc_prob: np.ndarray, number_of_probs_to_move: int
+) -> Tuple[List[int], np.ndarray]:
+    """
+    ``number_of_probs_to_move`` rows in the beginning are removed from ``acc_prob``. From every remove row the label
+    with the largest probability is selected and appended to ``pred``.
+    Args:
+        pred: list with ready label indices for a query
+        acc_prob: numpy array of shape ``[number_of_words_for_which_probabilities_are_accumulated, number_of_labels]``
+        number_of_probs_to_move: int
+    Returns:
+        pred: list with ready label indices for a query
+        acc_prob: numpy array of shape
+            ``[number_of_words_for_which_probabilities_are_accumulated - number_of_probs_to_move, number_of_labels]``
+    """
+    if number_of_probs_to_move > acc_prob.shape[0]:
+        raise ValueError(
+            f"Not enough accumulated probabilities. Number_of_probs_to_move={number_of_probs_to_move} "
+            f"acc_prob.shape={acc_prob.shape}"
+        )
+    if number_of_probs_to_move > 0:
+        pred = pred + list(np.argmax(acc_prob[:number_of_probs_to_move], axis=-1))
+    acc_prob = acc_prob[number_of_probs_to_move:]
+    return pred, acc_prob
+
+
+def _update_accumulated_probabilities(acc_prob: np.ndarray, update: np.ndarray) -> np.ndarray:
+    """
+    Args:
+        acc_prob: numpy array of shape ``[A, L]``
+        update: numpy array of shape ``[A + N, L]``
+    Returns:
+        numpy array of shape ``[A + N, L]``
+    """
+    acc_prob *= update[: acc_prob.shape[0]]
+    acc_prob = np.concatenate([acc_prob, update[acc_prob.shape[0] :]], axis=0)
+    return acc_prob
+
+
+def _remove_margins(tensor, margin_size, keep_left, keep_right):
+    if not keep_left:
+        tensor = tensor[margin_size + 1 :]  # remove left margin and CLS token
+    if not keep_right:
+        tensor = tensor[: tensor.shape[0] - margin_size - 1]  # remove right margin and SEP token
+    return tensor
+
+
+def apply_punct_capit_predictions(cfg: DictConfig, query: str, punct_preds: List[int], capit_preds: List[int]) -> str:
```

Comment:
move to the model class or create a utils file inside https://github.com/NVIDIA/NeMo/tree/main/nemo/collections/nlp/models/token_classification